library/include/rocwmma/internal/transforms_impl.hpp

```diff
@@ -211,6 +226,284 @@ namespace rocwmma
         }
     };
 
+    template <>
+    struct AosToSoa<16, 16>
+    {
+        constexpr static uint32_t VW       = 16;
+        constexpr static uint32_t VecSize  = 16;
+        constexpr static uint32_t BlockDim = 16;
+
+        template <typename DataT>
+        ROCWMMA_DEVICE constexpr static inline auto exec(VecT<DataT, VecSize> const& v)
+        {
+            using PackUtil = PackUtil<DataT>;
+
+            // Step 1 : Unpack groups of 1
+            auto result = unpackLoHi1(v);
+
+            // Step 2 : Unpack groups of 2
+            result = unpackLoHi2(result);
+
+            // Step 3 : Unpack groups of 4
+            result = unpackLoHi4(result);
+
+            // Step 4 : Unpack groups of 8
+            result = unpackLoHi8(result);
+
```

Comment:
Missing a gather?

Reply:
The use of UnpackLoHi1 leads to a state where no gather is necessary.